src/schema-association-service-1espt.ts

```diff
@@ -0,0 +1,92 @@
+                const config = vscode.workspace.getConfiguration('azure-pipelines')
+                config.update('1ESPipelineTemplatesSchemaFile', undefined, vscode.ConfigurationTarget.Workspace); // disable the 1ESPT schema configuration
+                vscode.window.showInformationMessage(Messages.disabled1ESPTSchemaAsADOOrgNotContains1ESPT)
+                return undefined; // 1ESPT repo not found
+            }
+            const repository = repositories.find(repo => repo.name === "1ESPipelineTemplates");
+            
+            // Using getItem from GitApi: getItem(repositoryId: string, path: string, project?: string, scopePath?: string, recursionLevel?: GitInterfaces.VersionControlRecursionType, includeContentMetadata?: boolean, latestProcessedChange?: boolean, download?: boolean, versionDescriptor?: GitInterfaces.GitVersionDescriptor, includeContent?: boolean, resolveLfs?: boolean, sanitize?: boolean): Promise<GitInterfaces.GitItem>;
+            var schemaFile = await gitApi.getItem(repository.id, "schema/1espt-base-schema.json", "1ESPipelineTemplates", undefined, undefined, true, true, true, undefined, true, true)
+
+            const schemaContent = schemaFile.content;
+            const schemaUri = Utils.joinPath(context.globalStorageUri, '1ESPTSchema', `${organizationName}-1espt-schema.json`);
+            await vscode.workspace.fs.writeFile(schemaUri, Buffer.from(schemaContent));
+            return schemaUri;
+        }
+        else
+        // if user is signed in with account other than microsoft, then disable 1ESPT schema and delete the 1ESPT schema file
+        {
+            const config = vscode.workspace.getConfiguration('azure-pipelines', workspaceFolder)
+            config.update('1ESPipelineTemplatesSchemaFile', undefined, vscode.ConfigurationTarget.Workspace);
+            await vscode.workspace.fs.delete(Utils.joinPath(context.globalStorageUri, '1ESPTSchema'), { recursive: true })
+        }
+    }
+    catch (error) {
+        logger.log(`Error : ${error} while fetching 1ESPT schema for org: ${organizationName}  : `, 'SchemaDetection');
+    }
+    return undefined;
+}
+
+export async function getCached1ESPTSchemaInformation(context: vscode.ExtensionContext, organizationName: string, session: AzureSession, oneesptSchemaEnabled: boolean, lastUpdated1ESPTSchema: Map<string,Date>, seen1ESPTOrganizations: Set<string>): Promise<[URI, boolean]> {
+    var skipOrgSpecificCachedSchema = false
+    var validSessionToEnable1ESPTSchema = false;
+    if (seen1ESPTOrganizations.has(organizationName)) {
+        const schemaUri1ESPT = Utils.joinPath(context.globalStorageUri, '1ESPTSchema', `${organizationName}-1espt-schema.json`);
+        validSessionToEnable1ESPTSchema = session.userId.endsWith("@microsoft.com") || session.userId.endsWith(".microsoft.com")
+
+        try
+        {
+            // check if 1ESPT schema file exists
+            var schemaFileExists = await vscode.workspace.fs.readFile(schemaUri1ESPT);
```

Comment:
If you're just looking for existence, `.stat` might be a better choice to avoid reading the entire file. (It is annoying how there's no exists() method...)
Also, consider that both readFile and stat will throw if the file doesn't exist, so you really don't need the if-check (or even the return value).

Reply:
got it. fixed it. Thanks!